core/src/main/java/org/infinispan/remoting/MIMECacheEntry.java

```diff
@@ -47,4 +55,35 @@ public boolean equals(Object o) {
    public int hashCode() {
       return 31 * (contentType != null ? contentType.hashCode() : 0) + (data != null ? Arrays.hashCode(data) : 0);
    }
+
+   public static class Externalizer extends AbstractExternalizer<MIMECacheEntry> {
+
+      @Override
+      public Set<Class<? extends MIMECacheEntry>> getTypeClasses() {
+         return Util.<Class<? extends MIMECacheEntry>>asSet(MIMECacheEntry.class);
+      }
+
+      @Override
+      public void writeObject(ObjectOutput out, MIMECacheEntry obj) throws IOException {
+         out.writeUTF(obj.contentType);
+         out.writeInt(obj.data.length);
+         out.write(obj.data);
```

Comment:
Why specifically? That method is calling this signature:

`write(byte b[])`

And that signature is just fine as is, it won't require any type lookups.